daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/TypeCalculator.scala

```diff
@@ -183,9 +146,9 @@ class KeysetValueTypeCalculatorOrdered(
   override def inputTypeCalc(
     x: DataValuePrimitive,
     xType: NodeInfo.Kind,
-  ): (DataValuePrimitiveNullable, Maybe[Error]) = {
+  ): Either[Error, DataValuePrimitiveNullable] = {
```

Comment:
Ah, so you also changed this return type from tuple to Either. That's cleaner also.